### PR TITLE
feat: Support for availability radio buttons

### PR DESCRIPTION
**Pull request by arnest00**
### Description
- Create Availability component

### Spec
See Story: [192](https://sparkbox.atlassian.net/browse/FSA22V1-192)

### Validation
* [x] This PR has code changes, and our linters still pass.
* [ ] This PR affects production code, so it was browser tested.
* [ ] Changes are browser tested. This includes functionality, accessibility, responsiveness, and design.

#### To Validate
1. Pull down `feat--availability-filter` branch.
2. Run `npm install`.
3. Confirm all tests pass.
4. Confirm all linters pass.
5. Open deploy preview and toggle between 'Available' and 'Unavailable.' There should be 11 available team members and 1 unavailable team member.

### CODE REVIEW

File: components/Availability.jsx [hunk 1]
```diff
@@ -0,0 +1,34 @@
+import { useState } from 'react';
+import { availabilityOptions, useFilterContext } from '../context/FilterContext';
+import FilterToggle from './FilterToggle';
+
+export default function Availability() {
+  const [checkedValue, setCheckedValue] = useState(availabilityOptions.AVAILABLE);
```
Comment (arnest00) [inline components/Availability.jsx:6]: To ensure React DOM and real DOM are congruous.

File: components/Availability.jsx [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    setCheckedValue(target.value);
+    setAvailability(target.value);
+  };
+
+  return (
+    <>
+      <FilterToggle
+        type="radio"
+        label="Available"
+        value={availabilityOptions.AVAILABLE}
```
Comment (mdt2) [inline components/Availability.jsx:19]: I love that these are constants 👏

File: components/OverviewFilters.jsx [hunk 3]
```diff
@@ -38,18 +39,7 @@ export default function OverviewFilters() {
       </fieldset>
       <fieldset className="cmp-overview-filters__group">
```
Comment (mdt2) [inline components/OverviewFilters.jsx:40]: The more I look at this `fieldset` the more I think it should be within the Availability component... The `fieldset` and `legend` group together and give information about the radio buttons, and the buttons wouldn't be accessible without them. So my "future proofing" thought is that we shouldn't even provide an opportunity for the `fieldset` and `legend` to be missed when using the Availability component. What are your thoughts?

**Review by mdt2 — COMMENTED** [review]
This is looking great! It's keyboard and VoiceOver accessible 🎉 Just left a couple comments, let me know what you think.

File: components/Availability.jsx [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+        name="av-buttons"
+        checked={(checkedValue === availabilityOptions.AVAILABLE)}
+        onChange={(e) => handleChange(e)}
+      />
+      <FilterToggle
+        type="radio"
+        label="Unavailable"
```
[the remark this answers is not in the document]
Comment (arnest00) [reply components/Availability.jsx:26]: That makes sense to me! We can just `text-transform: capitalize` the label class in FilterToggle in this case.

File: styles/components/_fieldset-group.scss [hunk 5]
```diff
@@ -0,0 +1,14 @@
+.cmp-fieldset-group {
```
Comment (mdt2) [inline styles/components/_fieldset-group.scss:1]: Nice renaming here to keep it relevant to the component 👍

File: components/Availability.jsx [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+        type="radio"
+        label={availabilityOptions.AVAILABLE}
+        value={availabilityOptions.AVAILABLE}
+        name="av-buttons"
+        checked={(checkedValue === availabilityOptions.AVAILABLE)}
+        onChange={(e) => handleChange(e)}
+      />
+      <FilterToggle
+        type="radio"
+        label={availabilityOptions.UNAVAILABLE}
```
Comment (mdt2) [inline components/Availability.jsx:27]: 🎉

File: components/Role.jsx [hunk 7]
```diff
@@ -0,0 +1,39 @@
+import FieldsetGroup from './FieldsetGroup';
+import FilterToggle from './FilterToggle';
+
+export default function Availability() {
+  return (
+    <FieldsetGroup legend="role">
+      <FilterToggle
+        type="checkbox"
+        label="Developers"
```
Comment (mdt2) [inline components/Role.jsx:9]: I'm guessing these don't have constants already declared? I think that's ok since they only seem to be used once 👍

**Review by mdt2 — DISMISSED** [review]
Changes look great! I think this one is good to go.

r2BtghAUTmpP2

**Comment by arnest00** [comment]
@theogainey Just as a heads up, there are some changes here that would affect #39 if we decide to keep them, namely extracting components from OverviewFilters and some CSS class rearanging.

**Review by theogainey — CHANGES_REQUESTED** [review]
Overall solid work but there is a one thing I think would make this code much better. The that thing  is that  pushing the Radio buttons down from OverviewFilters to FieldsetGroup to Availability makes the logic and styles kind of hard to follow in my opinion. It would be one thing if the possibility of a bunch a radio buttons existed but we can safely assume that at most 1 more will be added so why not just have these buttons live in OverviewFilters. Really the only thing you would need to add is  
```
const { availability, setAvailability } = useFilterContext();
const handleChange = ({ target }) => {
     setAvailability(target.value);
   };
```
then attach the handle to the buttons.

Feel free to share any disagreement.

File: components/Availability.jsx [hunk 8]
```diff
@@ -0,0 +1,35 @@
+import { useState } from 'react';
+import { availabilityOptions, useFilterContext } from '../context/FilterContext';
+import FieldsetGroup from './FieldsetGroup';
+import FilterToggle from './FilterToggle';
+
+export default function Availability() {
+  const [checkedValue, setCheckedValue] = useState(availabilityOptions.AVAILABLE);
```
[the remark this answers is not in the document]
Comment (arnest00) [reply components/Availability.jsx:7]: That was originally because when I'd used `availability` from context, the state was updating, but the DOM wasn't. Although something must have been wonky in my original implementation because it's working fine with this edit now.

**Comment by arnest00** [comment]
> Overall solid work but there is a one thing I think would make this code much better. The that thing is that pushing the Radio buttons down from OverviewFilters to FieldsetGroup to Availability makes the logic and styles kind of hard to follow in my opinion. It would be one thing if the possibility of a bunch a radio buttons existed but we can safely assume that at most 1 more will be added so why not just have these buttons live in OverviewFilters.

My rationale was separation of responsibilities. Instead of OverviewFilters having the handlers for both Availability and Roles, 
they'd live in their own component.

I extracted out FieldsetGroup so we wouldn't have to redo the `fieldset` and `legend` styling for both components as well.

There's also the rest of the work we'll be doing with this section of the app. For [193](https://sparkbox.atlassian.net/browse/FSA22V1-193), we can stick SelectControl (for projects), Availability, and Roles into OverviewFilters since they'll all need to be in the same container.

**Comment by theogainey** [comment]
One last pitch for doing it the way I am suggesting. Raking into account the current work I have in review,  this is what `<OverviewFilters >` or `<Filters>` if it needs to be renamed  would look like. 
```
/*
context and functions
*/
<div className="cmp-overview-filters">
      <SelectControl .../>
      <fieldset className="cmp-overview-filters__group">
        <legend className="cmp-overview-filters__group-label">Role</legend>
        {uniqueRoles && uniqueRoles.map((role) => (
          <FilterToggle
            key={role}
            type="checkbox"
            label={role}
            value={role}
            name="role"
            onChange={() => toggleRole(role)}
          />
        ))}
      </fieldset>
      <fieldset className="cmp-overview-filters__group">
        <legend className="cmp-overview-filters__group-label">Availability</legend>
        <FilterToggle
          type="radio"
          label"="Available"
          value="available"
          name="av-buttons"
         checked={(availability === availabilityOptions.AVAILABLE)}
         onChange={(e) => handleChange(e)}
        />
        <FilterToggle
          type="radio"
          label="Unavailable"
          value="unavailable"
          name="av-buttons"
         checked={(availability === availabilityOptions.UNAVAILABLE)}
         onChange={(e) => handleChange(e)}
        />
      </fieldset>
    </div>
```

This is a very manageable component with 4 less files to work with and all the SCSS for the component, minus the styles for individual buttons and select, will live in 1 file not 2.

**Comment by mdt2** [comment]
@arnest00 @theogainey I'm  loving this dialogue! This might be a good one to take to the team in the #dev channel if you two are comfortable with that. I'm sure that folks will have opinions and examples that may be helpful in making a final decision 😄

**Review by theogainey — APPROVED** [review]
Nesting won.
